documentation/DEVELOPER-GUIDELINES.md

```diff
@@ -35,7 +35,11 @@ In general, the [Rust API Guidelines](https://rust-lang.github.io/api-guidelines
   - The `ConfigError` enum should be separate from other `Error` enums used by the driver.
   - The driver should implement `fn apply_config(&mut self, config: &Config) -> Result<(), ConfigError>`.
   - In case the driver's configuration is infallible (all possible combinations of options are supported by the hardware), the `ConfigError` should be implemented as an empty `enum`.
-  - Configuration structs should derive `procmacros::BuilderLite` in order to automatically implement the Builder Lite pattern for them.
+  - Configuration structs should
+    - Derive `procmacros::BuilderLite` in order to automatically implement the Builder Lite pattern for them.
```

Comment:
We should probably relax this to using the pattern, but the macro should be optional/preferred. We need to allow for manual additions like register value precalcuation.